From 2552ccc21750fca138e9d10f833932ff379ba684 Mon Sep 17 00:00:00 2001
From: Matti Paasto <matti.paasto@intel.com>
Date: Mon, 22 May 2017 13:16:23 +0300
Subject: [PATCH] Fix static code analysis finding in loader Out of bounds
 access can happen in loader due to usage of wrong struct

Signed-off-by: Matti Paasto <matti.paasto@intel.com>
---
 loader/loader.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/loader/loader.c b/loader/loader.c
index bff79c1..9ae3d56 100644
--- a/loader/loader.c
+++ b/loader/loader.c
@@ -1151,7 +1151,7 @@ loader_add_to_dev_ext_list(const struct loader_instance *inst,
     }
 
     memcpy(&ext_list->list[idx].props, props,
-           sizeof(struct loader_dev_ext_props));
+           sizeof(VkExtensionProperties));
     ext_list->list[idx].entrypoint_count = entry_count;
     ext_list->list[idx].entrypoints =
         loader_instance_heap_alloc(inst, sizeof(char *) * entry_count,
-- 
2.7.4

